src/pcapi/core/offers/api.py

```diff
@@ -458,3 +463,55 @@ def update_mediation(mediation: Mediation, is_active: bool) -> Mediation:
         redis.add_offer_id(client=app.redis_client, offer_id=mediation.offerId)
 
     return mediation
+
+
+def create_mediation_v2(
+    user: User,
+    offer: Offer,
+    credit: str,
+    image_as_bytes: bytes,
+    crop_params: tuple = None,
+) -> Mediation:
+    # checks image type, min dimensions
+    validation.check_image(image_as_bytes)
+
+    existing_mediations = mediation_queries.get_mediations_for_offers([offer.id])
+
+    mediation = Mediation(
+        author=user,
+        offer=offer,
+        credit=credit,
+    )
+    # `create_thumb()` requires the object to have an id, so we must save now.
+    repository.save(mediation)
+
+    try:
+        create_thumb(mediation, image_as_bytes, image_index=0, crop_params=crop_params)
+
+    except Exception as exc:
+        app.logger.exception("An unexpected error was encountered during the thumbnail creation: %s", exc)
+        # I could not use savepoints and rollbacks with SQLA
+        repository.delete(mediation)
+        raise ThumbnailStorageError
+
+    else:
+        mediation.thumbCount = 1
+        db.session.commit()
+        # cleanup former thumbnails and mediations
+        for old_mediation in existing_mediations:
```

Comment:
```suggestion
        for previous_mediation in existing_mediations:
```
micro suggestion mais pas important